latex/epoch.tex

```diff
@@ -77,6 +89,65 @@
 to assign a refund to each pool key in the set of pools scheduled for retirement
 at the epoch boundary.
 
+%%
+%% Figure - Functions for Stake Distribution
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{baseStake} \in \powerset{\TxOut} \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{baseStake}~{txouts} = \\
+      & ~~ \{(hk, c) \mid a\in\Addr_{base} \land (a, c) \in txouts \land \fun{stakeHK}~a = hk \}
+      \nextdef
+      & \fun{ptrStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{ptrStake}~{txouts}~{ptrs} = \\
+      & ~~ \{(hk, c) \mid a\in\Addr_{ptr} \land (a, c) \in txouts \land (\fun{stakePtr}~a\mapsto hk \in ptrs \}
+      \nextdef
+      & \fun{stake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{stake}~{txouts}~{ptrs} = (\fun{baseStake}~{txouts}) \cup (\fun{ptrStake}~{txouts}~{ptrs})
+      \nextdef
+      & \fun{isActive} \in \HashKey \to \Allocs \to (\HashKey \mapsto \HashKey) \to \Allocs \to \mathbb{B}
+      & \text{active stake} \\
+      & \isActive{v_{sk}}{stkeys}{delegs}{stpools} = \exists \var{v_{p}}\\
+      & ~~ \var{v_{sk}} \in \dom stkeys
+              \land \var{v_{st}}\mapsto\var{v_{p}} \in delegs
+              \land \var{v_{p}} \in \dom \var{stpools}
+      \nextdef
+      & \fun{activeStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey) \to \Allocs\\
+      & ~~\to (\HashKey \mapsto \HashKey) \to \Allocs \to (\VKey \mapsto \Coin)
+      & \text{active stake holdings} \\
+      & \activeStake{txouts}{ptrs}{stkeys}{delegs}{stpools} = \\
+      & ~~ \left\{
+             hk\mapsto \sum_{hk\mapsto c}c
+             ~\Big\vert~
+             \begin{array}{r}
+             (\isActive{hk}{stkeys}{delegs}{stpools}) \\
+             \land (hk, c) \in (\fun{stake}~{txouts}~{ptrs})
+             \end{array}
+           \right\}
+      \nextdef
+      & \fun{totalStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey) \to \Allocs\\
```

Comment:
in "(hk, c) ∈ (stake txouts ptrs)", hk is a hash key, and the result type is VKey \to Coin
How should we reconcile this?